packages/evolution-legacy/src/components/survey/Survey.js

```diff
@@ -58,6 +60,14 @@ export class Survey extends React.Component {
     const pathSectionParentSection = pathSectionShortname && this.surveyContext.sections[pathSectionShortname] ? this.surveyContext.sections[pathSectionShortname].parentSection : null;
     const { state } = this.context;
     this.props.startSetInterview(existingActiveSection || pathSectionParentSection, surveyUuid, state.status === 'entering' && Object.keys(state.responses).length > 0 ? state.responses : undefined);
+
+    // Override the window.console functions to log errors and warnings to the server. It is here because it's the top most component with an interview to associate with
+    overrideConsoleLogs({ getInterviewId: () => this.props.interview ? this.props.interview.id : undefined });
+  }
+
+  componentWillUnmount() {
+    // Reset the console functions to their original values
+    restoreConsoleLogs();
```

Comment:
symetry: since the log reporting works only for logged in users, when the component unmounts (for example if the user logs out), we want to restore the original behavior to avoid useless potential 401 requests.

Reply:
I guess I'll refer to my other comment about have non lgged log